packages/site/cms-site/src/blocks/helpers/useIsElementVisible.tsx

```diff
@@ -0,0 +1,20 @@
+import { RefObject, useEffect, useState } from "react";
+
+export const useIsElementVisible = (ref: RefObject<HTMLDivElement>) => {
```

Comment:
Naming: useIsElementInViewport would probably be better. An element can still be hidden using `display: none`.

BTW, nice implementation using IntersectionObserver 👏🏼

Reply:
Thank you :) 

Naming changed here: 0ebb7d3e13ed3706e4fc1fc3f05b43a8e627b919